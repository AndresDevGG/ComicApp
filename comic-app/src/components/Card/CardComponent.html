<div class="card">
    <div class="header">
        <h1 class="title">{{title}}</h1>
        <i @click="setView()" class="ri-arrow-go-back-line pointer reverse"></i>
    </div>
    <div class="front" v-bind:class="{'d-none': !isFront}">
        <slot name="front"></slot>
    </div>
    <div class="back" v-bind:class="{'d-none': isFront}">
        <slot name="back"></slot>
    </div>

</div>