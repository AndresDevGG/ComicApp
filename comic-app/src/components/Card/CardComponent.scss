@import './../../assets/colors.scss';
.card {
    width: 700px;
    height: 500px;
    border-radius: 10px;
    background-color: $green-dark;
    box-shadow: 0px 3px 18px 1px $green-dark;
    padding: 1rem;
    .header {
        display: flex;
        justify-content: space-between;
        align-items: center;
        margin-bottom: 1rem;
    }
    
    .title {
        margin: 0;
    }

    .reverse {
        font-size: 20px;
    }

    .reverse:hover {
        transition: .2s;
        font-size: 25px;
    }

    
}

.d-none {
    display: none;
}


@media (prefers-color-scheme: light) {
    .card {
        background-color: #fff;
        box-shadow: 0px 3px 18px 1px rgb(187 187 187 / 61%);
    }
}