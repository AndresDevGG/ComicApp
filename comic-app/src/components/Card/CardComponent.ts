import { defineComponent, ref } from 'vue';

export default defineComponent({
    name: "Card",
    props: {
        title: String
    },
    setup() {
        let isFront = ref(true);

        const setView = () => isFront.value = !isFront.value;

        return {
            isFront,
            setView
        }
    }    
});
