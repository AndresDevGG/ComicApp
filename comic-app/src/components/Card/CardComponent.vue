<script lang="ts" src="./CardComponent"></script>

<template src="./CardComponent.html"></template>
<style lang="scss" src="./CardComponent.scss"></style>
