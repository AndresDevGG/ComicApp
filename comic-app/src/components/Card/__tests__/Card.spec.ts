import { describe, expect, it, test } from 'vitest'

import  Card from './../CardComponent'
import { mount } from '@vue/test-utils'

describe('Card', () => {
    const title: string = 'This is a title!';
    
    const wrapper = mount(Card, {
        props: {
            title: title
        }
    })

    it('Should render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    });

    it('Should render title', async () => {
        expect(wrapper.html().includes(title))
    });
})