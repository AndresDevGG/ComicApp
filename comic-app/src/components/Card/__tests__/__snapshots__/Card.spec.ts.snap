// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Card > Should render correctly 1`] = `""`;
